src/core/api/utils.go

```diff
@@ -278,3 +280,22 @@ func repositoryExist(name string, client *registry.Repository) (bool, error) {
 	}
 	return len(tags) != 0, nil
 }
+
+func requireLock(key string) (*redis.Mutex, error) {
```

Comment:
There is already a same function for image quota? Can we use the same one?

Reply:
Move it to the `common/utils/redis` package, it's a helper function to get lock of the key using the default redis connection, and require the lock by the same function for image quota.